examples/rosettacode/24_game.nit

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env nit
+#
+# This file is part of NIT ( http://www.nitlanguage.org ).
+# This program is public domain
+
+# Task: 24 game
+# SEE: <http://rosettacode.org/wiki/24_game>
+
+redef class Char 
+	fun is_op: Bool do return "-+/*".has(self)
+	fun is_dig: Bool do return [0..9].has(self.to_i)
+end
+
+# Export `operands` and `numbers` from string `operation`
+fun exportation(operation: String, numbers: Array[Int], operands: Array[Char]) do
+	var previous_char: nullable Char = null
+	var number: nullable Int = null
+	var negative = false
+
+	for i in operation.length.times do
+		var current_char = operation[i]
+		var current_int = current_char.to_i
+
+		if (previous_char == null or previous_char.is_op) and current_char == '-' then
+			negative = true
+			continue
+		end
+
+		if current_char.is_dig then
+			if number == null then
+				number = current_int
+			else
+				number = number * 10 + current_int
+			end
+		end
+
+		if negative and (current_char.is_op or i == operation.length - 1) then
+			number = number - number * 2
+			negative = false
+		end
+
+		if (current_char.is_op or i == operation.length - 1) and number != null then
+			numbers.add(number)
+			number = null
+		end
+
+		if not negative and current_char.is_op then
+			operands.add(current_char)
+		end
+		previous_char = current_char
+	end
+end
```

Comment:
Yeah, it's early in the morning and I still have no cue about what happened here.